src/AbstractApplication.php

```diff
@@ -62,6 +67,29 @@ public function __construct(Input $input = null, Registry $config = null)
 		$this->input = $input instanceof Input ? $input : new Input;
 		$this->config = $config instanceof Registry ? $config : new Registry;
 
+		// Create the container
```

Comment:
Should not the application take care about filling the container the way it is needed to run the application? IMO the service providers should be attached in the constructor (or at least early in the lifecycle) of the application and the bootup script should be only a couple of lines big. Having the app part of the global container, it is not possible to get the container inside the application.